libs/contrib/Data/Bool/Extra.idr

```diff
@@ -0,0 +1,28 @@
+module Data.Bool.Extra
+
+%access public export
+%default total
+
+andDelaySameNeutral : (x : Bool) -> x && Delay x = x
+andDelaySameNeutral False = Refl
+andDelaySameNeutral True = Refl
+
+andDelayFalseFalse : (x : Bool) -> x && Delay False = False
+andDelayFalseFalse False = Refl
+andDelayFalseFalse True = Refl
+
```

Comment:
Why do you have explicit `Delay`s here, don't logical ops automatically insert that?

Reply:
I assumed that the errors were due to the second argument being lazy, so I wrote it that way for clarity. This assumption may have been incorrect, so I'll do a little digging and revise if I was wrong.

Reply:
Turns out it was a regular old reduction issue. I tried strict versions of `&&` and `||` and had the same result. I'll rename the functions accordingly and fix their types.

Do you think it would be a bad idea to declare the equality in the type in reverse, to avoid using `sym` everywhere? I'm afraid it wouldn't be as semantically correct.

Reply:
Why would you use `sym`, are you rewriting `False` back to `x && False`?

Reply:
I don't think so... But each use in libs/contrib/Text/Parser.idr needs a `sym`.

Reply:
Ah, so you indeed are rewriting backwards: e.g., in `option` you convert `Grammar tok False a` to `Grammar tok (c && False) a` so that `<|>` could be applied. You could also use `replace (andFalseFalse c) (p <|> pure def)` but that would most likely require writing the `P` implcit term, which I feel is somewhat ugly.